docs/config/README.md

```diff
@@ -349,6 +350,29 @@ modules you explicitly add to the format will not be duplicated. Eg.
 format = '$all$directory$character'
 ```
 
+### Profiles
+
+Profiles can be used to set a key-value-map with alternate versions of the top-level `format` (the main prompt format string) and `right_format` (the same for the `right_prompt`). To quickly switch to a different profile of the current prompt, you can set the `STARSHIP_PROFILE` environment variable to the name of the profile, for instance by running `export STARSHIP_PROMPT=my_profile` or `$ENV:STARSHIP_PROMPT="my_profile"` in PowerShell. Alternatively, it can also be set via an `--profile=my_profile` flag for commands like `starship prompt` and `starship explain`.
+
+| Option         | Default | Description                                                                             |
+| -------------- | ------- | --------------------------------------------------------------------------------------- |
+| `format`       | `''`    | Configure the format of the prompt profile.                                             |
+| `right_format` | `''`    | [Enable the right prompt of the prompt profile.](/advanced-config/#enable-right-prompt) |
+
+#### Examples
+
+```toml
+# Define custom profiles
+[profiles]
+# Instead of a map you can also set a profile to a simple string instead
+# Starship includes the definition below for use with transient prompts
+transient = "$character"
+
+[profiles.as_map_with_right_format]
+format = "$character"
+right_format = "$git_branch"
+```
+
```

Comment:
Hello David,

While reading issue #4828, I was contemplating, where to add further documentation about the `[palettes]` option. In my opinion, the `Prompt` section of the _Configuration_ documentation is already quite extensive. Therefore, I'll add it to the _Advanced Documentation_ section.

Following my above reasoning, I would suggest to also put the _Profiles_ part to the _Advanced Configuration_ section.

Just my 2 cents
Mick

Reply:
I feel like the _Advanced Configuration_ docs aren't very visible, which is why I put it in the main config docs, but I will consider moving it.

Reply:
@davidkna, should the new `[profiles]` section already work in the v1.13.1 release, and you just add the documentation?

Reply:
Apologies for weighing in late. I am on mobile, but wanted to capture a thought that I can't currently confirm.

I agree @mickimnet, the advanced configuration docs are rather extensive, and generally amazing. To @davidkna's point, the whole reason for https://github.com/starship/starship/issues/4828 was the lack of concept linking between the color/pallette feature in the main docs and the advanced configuration.

We could likely satisfy both desires (to keep advanced topics where they are, but provide visibility) by including links to the advanced configuration topics in the basic documentation.

As a general practice, I think that this improves UX, indicates that desired capabilities already exist, and also implicitly let's users know that they're venturing into advanced territory - it also acts as an `apropos` of sorts to connect the advanced configuration to its (more searchable/navigable) basic counterparts.

TL;DR - if we leave the advanced config where it is and just link basic config to further reading in "advanced config" we may empower users who lack the intuition/context on where to find the info they need - without additional human intervention.